### PR TITLE
Read imp

### PR DESCRIPTION
**Pull request by obadrawi**
1-implement read functions
2- fix update functions
3- implement showing action status

### CODE REVIEW

File: OWNERS [hunk 1]
```diff
@@ -1,6 +1,7 @@
 approvers:
  - clebs
  - suleymanakbas91
+ - igor-karpukhin
```
Comment (clebs) [inline OWNERS]: Please remove this line.

File: CODEOWNERS [hunk 2]
```diff
@@ -7,5 +7,6 @@
 # These are the default owners for the whole content of the `terraform-provider-gardener` repository. The default owners are automatically added as reviewers when you open a pull request, unless different owners are specified in the file.
 * @a-thaler @hisarbalik @lilitgh @suleymanakbas91 @clebs
 
+
```
Comment (clebs) [inline CODEOWNERS]: Please remove this line.

File: shoot/gcp.go [hunk 3]
```diff
@@ -118,3 +119,54 @@ func getGCPWorkers(d *schema.ResourceData) []gardener_types.GCPWorker {
 	}
 	return resultWorkers
 }
+
+func updateGCPSpec(d *schema.ResourceData, gcpSpec *gardener_types.GCPCloud) *gardener_types.GCPCloud {
```
Comment (clebs) [inline shoot/gcp.go]: Usually if a pointer is given as output argument, changes made in the function are available to the caller via the pointer, no need to return it as it only confuses the caller.

Please adjust this function and all other similar ones in this package.

**Review by clebs — CHANGES_REQUESTED** [review]

File: shoot/gcp.go (hunk 3 above)
Comment (obadrawi) [reply shoot/gcp.go]: now the function doesn't return anything

File: CODEOWNERS (hunk 2 above)
Comment (obadrawi) [reply CODEOWNERS]: removed

File: OWNERS (hunk 1 above)
Comment (obadrawi) [reply OWNERS]: removed

**Review by clebs — APPROVED** [review]

**Pull request merged**